### ISSUE TITLE
VIDIOC_QBUF CAPTURE errors while playing iptv-all mediaset

### ISSUE THREAD

**Issue by EwanRoycroft** [bbc/chromium#29]
### Description
When using the `iptv-all` mediaset with the [PX VOD](https://github.com/bbc/px-vod) player, the V4L2 decoder will report many decoder errors soon after playback begins.

SMP seems to try switching between different CDNs for the DASH stream, each encountering decoder errors immediately, before giving up. The V4L2 decoder reports `EFAULT` errors when queuing CAPTURE buffers.

This only seems to occur for DASH representations <720p, and only for content played in SMP. No errors are encountered when playing R&D testcard streams using the [SMP "href" feature](https://confluence.dev.bbc.co.uk/display/mp/Loading+a+Playlist+with+an+object).

### More information
| Attribute | Value |
| --- | --- |
| **Branch** | `px-rpi-v89` |
| **Version** | 89.0.4389.72 |
| **Build** | Release |
| **Platform** | Ozone-DRM |
| **Target** | `video_decode_accelerator_tests` |
| **Device** | Raspberry Pi 4B (4GB) |
| **OS** | Raspbian 10 "buster" |
| **Kernel** | 5.10.17-v7l+ |

### Steps to reproduce
Play a video using PX VOD. Ensure that it is using the [`tv-mediaset` branch](https://github.com/bbc/px-vod/tree/tv-mediaset).

#### GN arguments
```
use_ozone = true
ozone_platform_x11 = false
ozone_platform_drm = true
ozone_platform_headless = false
ozone_platform = "drm"
use_vc4_minigbm = true

ffmpeg_branding = "Chrome"
proprietary_codecs = true
use_v4l2_codec = true
use_v4l2_non_standard = false
use_v4l2_stateless_decoder = false

target_os = "chromeos"
target_cpu = "arm"
use_sysroot = true
target_sysroot = "//build/linux/raspbian_arm-sysroot"
enable_linux_installer = true

google_api_key = "no"
google_default_client_id = "no"
google_default_client_secret = "no"

is_debug = false
is_component_build = false
enable_nacl = false
symbol_level = 0
blink_symbol_level = 0
```

#### Command
```shell
./chrome --force-system-compositor-mode --enable-gpu-rasterization --disable-explicit-dma-fences
```

### Terminal output
```
2021-08-02T15:03:23.511247Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 main and fourcc: H264
2021-08-02T15:03:23.552587Z VERBOSE3 chrome[1367:1499]: [v4l2_device.cc(2274)] GetCtrl(): Failed to get control: Inappropriate ioctl for device (25)
2021-08-02T15:03:23.553286Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder_backend_stateful.cc(555)] ChangeResolution(): Using default minimum number of CAPTURE buffers
2021-08-02T15:03:23.555914Z VERBOSE1 chrome[1367:1499]: [v4l2_device.cc(995)] SetVisibleRect(): visible_rect=0,0 704x396
2021-08-02T15:03:23.557758Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(388)] SetupOutputFormat(): buffer modifier: 0
2021-08-02T15:03:23.577992Z VERBOSE3 chrome[1367:1499]: [v4l2_device.cc(2274)] GetCtrl(): Failed to get control: Inappropriate ioctl for device (25)
2021-08-02T15:03:23.578104Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder_backend_stateful.cc(555)] ChangeResolution(): Using default minimum number of CAPTURE buffers
2021-08-02T15:03:23.582438Z VERBOSE1 chrome[1367:1499]: [v4l2_device.cc(995)] SetVisibleRect(): visible_rect=0,0 704x396
2021-08-02T15:03:23.582554Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(388)] SetupOutputFormat(): buffer modifier: 0
2021-08-02T15:03:27.222066Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 high and fourcc: H264
2021-08-02T15:03:27.235369Z VERBOSE1 chrome[1367:1499]: [v4l2_device.cc(1245)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-08-02T15:03:27.235501Z ERROR chrome[1367:1499]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-08-02T15:03:27.235605Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.

[ Repeats many times ]

2021-08-02T15:03:27.353974Z VERBOSE1 chrome[1367:1499]: [v4l2_device.cc(1245)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-08-02T15:03:27.354054Z ERROR chrome[1367:1499]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-08-02T15:03:27.354092Z VERBOSE1 chrome[1367:1499]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.
2021-08-02T15:03:27.579834Z VERBOSE2 chrome[1367:1505]: [v4l2_video_decoder.cc(93)] V4L2VideoDecoder():
2021-08-02T15:03:27.582242Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 main and fourcc: H264
2021-08-02T15:03:27.629851Z VERBOSE3 chrome[1367:1505]: [v4l2_device.cc(2274)] GetCtrl(): Failed to get control: Inappropriate ioctl for device (25)
2021-08-02T15:03:27.629996Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder_backend_stateful.cc(555)] ChangeResolution(): Using default minimum number of CAPTURE buffers
2021-08-02T15:03:27.633212Z VERBOSE1 chrome[1367:1505]: [v4l2_device.cc(995)] SetVisibleRect(): visible_rect=0,0 704x396
2021-08-02T15:03:27.634421Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder.cc(388)] SetupOutputFormat(): buffer modifier: 0
2021-08-02T15:03:27.650526Z VERBOSE3 chrome[1367:1505]: [v4l2_device.cc(2274)] GetCtrl(): Failed to get control: Inappropriate ioctl for device (25)
2021-08-02T15:03:27.651153Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder_backend_stateful.cc(555)] ChangeResolution(): Using default minimum number of CAPTURE buffers
2021-08-02T15:03:27.653551Z VERBOSE1 chrome[1367:1505]: [v4l2_device.cc(995)] SetVisibleRect(): visible_rect=0,0 704x396
2021-08-02T15:03:27.653669Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder.cc(388)] SetupOutputFormat(): buffer modifier: 0
2021-08-02T15:03:28.474500Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder.cc(218)] Initialize(): Using a stateful API for profile: h264 high and fourcc: H264
2021-08-02T15:03:28.489887Z VERBOSE1 chrome[1367:1505]: [v4l2_device.cc(1245)] QueueBuffer(): (CAPTURE_MPLANE) VIDIOC_QBUF failed: Bad address (14)
2021-08-02T15:03:28.490018Z ERROR chrome[1367:1505]: [v4l2_video_decoder_backend_stateful.cc(308)] Error while queueing output buffer!
2021-08-02T15:03:28.490063Z VERBOSE1 chrome[1367:1505]: [v4l2_video_decoder.cc(717)] SetState(): Error occurred, stopping queues.

[ Repeats many times ]
```

**Comment by EwanRoycroft** [comment]
Buffer errors in #25 are also caused by `EFAULT`. Could these errors be due to Chromium reinitialising or destroying the decoder?

**Comment by EwanRoycroft** [comment]
This can likely be closed with #32.
